docs/generating-a-provider.md

```diff
@@ -0,0 +1,391 @@
+   a boilerplate code in file `internal/clients/${ProviderNameLower}.go` which
+   takes care of properly fetching secret data referenced from `ProviderConfig`
+   resource.
+
+   For our GitHub provider, we need to check [Terraform documentation for provider
+   configuration] and provide the keys there:
+
+   ```go
+   const (
+     keyBaseURL = "base_url"
+     keyOwner = "owner"
+     keyToken = "token"
+
+     // GitHub credentials environment variable names
+     envToken = "GITHUB_TOKEN"
+   )
+
+   func TerraformSetupBuilder(version, providerSource, providerVersion string) terraform.SetupFn {
+     ...
+     // set provider configuration
+     ps.Configuration = map[string]interface{}{}
+     if v, ok := githubCreds[keyBaseURL]; ok {
+         ps.Configuration[keyBaseURL] = v
+     }
+     if v, ok := githubCreds[keyOwner]; ok {
+         ps.Configuration[keyOwner] = v
+     }
+     // set environment variables for sensitive provider configuration
+     ps.Env = []string{
+         fmt.Sprintf(fmtEnvVar, envToken, githubCreds[keyToken]),
+     }
+     return ps, nil
+   }
+   ```
+
+6. Before generating all resources that the provider has, let's go step by step
+   and only start with generating `github_repository` and `github_branch`
+   resources.
+
+   To limit the resources to be generated, we need to provide an include list
+   option with `tjconfig.WithIncludeList` in file `config/provider.go`:
+
+   ```go
+   pc := tjconfig.NewProvider(resourceMap, resourcePrefix, modulePath,
+       tjconfig.WithDefaultResourceFn(defaultResourceFn),
+       tjconfig.WithIncludeList([]string{
+           "github_repository$",
+           "github_branch$",
+       }))
+   ```
+
+7. Finally, we would need to add some custom configurations for these two
+   resources as follows:
+
+   ```bash
+   # Create custom configuration directory for whole repository group
+   mkdir config/repository
+   # Create custom configuration directory for whole branch group
+   mkdir config/branch
+   ```
+
+   ```bash
+   cat <<EOF > config/repository/config.go
+   package repository
+
+   import "github.com/crossplane-contrib/terrajet/pkg/config"
+
+   func Customize(p *config.Provider) {
+       p.AddResourceConfigurator("github_repository", func(r *config.Resource) {
+   
+           // we need to override the default group that terrajet generated for
+           // this resource, which would be "github"  
+           r.Group = "repository"
+       })
+   }
+   EOF
+   ```
+
+   ```bash
+   cat <<EOF > config/branch/config.go
+   package branch
+
+   import "github.com/crossplane-contrib/terrajet/pkg/config"
+
+   func Customize(p *config.Provider) {
+       p.AddResourceConfigurator("github_branch", func(r *config.Resource) {
+   
+           // we need to override the default group that terrajet generated for
+           // this resource, which would be "github" 
+           r.Group = "branch"
```

Comment:
```suggestion
           r.Group = "repos"
```

In the original [Github API](https://docs.github.com/en/rest/reference/repos#branches), that's the group used for both `repository` and `branch` resources, which is what we're following in [native provider-github](https://github.com/crossplane-contrib/provider-github/blob/main/apis/organizations/v1alpha1/register.go#L28), too. As I mentioned in another PR, I think we should tell people to follow the API instead of Terraform grouping so that migration/merge scenarios are less-breaking.

Reply:
I'll keep this as is not to confuse people following the guide here, which is also aligned what we agreed here: https://github.com/crossplane-contrib/provider-jet-gcp/pull/2#discussion_r749302880